python_interface/design/04_pyHorace_packaging.md

```diff
@@ -0,0 +1,61 @@
+# A description of the prototype pyHorace implementation
+**2021-1-3**
+
+
+# Background
+
+`pyHorace` will be a Python module which bundles together PACE programs written in Matlab for use by Python users without a Matlab license.
+
+This document describes the options for packaging it for distribution to users.
+
+
+# pyHorace packaging options
+
+There are a few ways to distribute `pyHorace` which we need to decide on:
+
+1. OS-specific packaging system
+2. The Python Package Index (PyPI)
+3. The Anaconda (`conda`) system
+
+Option 1 would be to create packages for specific OS (e.g. `msi` for Windows, `dmg` for Mac OS, `deb` for Debian/Ubuntu, `rpm` for Red Hat/CentOS etc.).
+These packages would be installed by users as they would a non-Python program.
+Options 2 and 3 are specifically for Python packages (or packages meant to be used with Python).
+
+A further decision is whether we should "bundle" the MCR with the `pyHorace` distribution or not.
+The MCR is a very large archive (~2.8GB for R2020b) because it basically contains the full Matlab distribution.
+If we want to distribute the MCR with the package, we are restricted to option 1 and self-hosting.
+This is because PyPI has a limit of 100MB per package and Anaconda has a limit of 1GB per package.
+
```

Comment:
```suggestion

Recently (since R2020a), Matlab introduced a command, [`compiler.package.installer`](https://uk.mathworks.com/help/compiler/compiler.package.installer.html) which can create an installer executable which will only download the components / toolboxes needed by the application. For `pyHorace` this reduces the download size to ~500MB, since we only need the core computation and Python components. It is not possible to bundle this reduced sized MCR with our application however - it seems the only way is to create an installer using `compiler.package.installer`.

```